x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateExtract.java

```diff
@@ -114,11 +115,20 @@ protected TypeResolution resolveType() {
         if (childrenResolved() == false) {
             return new TypeResolution("Unresolved children");
         }
-        TypeResolution resolution = isDate(children().get(0), sourceText(), TypeResolutions.ParamOrdinal.FIRST);
+        TypeResolution resolution = argumentTypesAreSwapped(
```

Comment:
super-duper nitpick/bikeshed: for me, the naming `argumentTypesAreSwapped` suggests "the arguments are swapped, so let's act accordingly", rather than "check if the arguments are swapped". Maybe consider renaming to `areArgumentTypesSwapped` or `checkIfArgumentTypesAreSwapped` or so.